libpod/container_internal_common.go

```diff
@@ -2203,6 +2205,17 @@ func (c *Container) getHostsEntries() (etchosts.HostEntries, error) {
 	switch {
 	case c.config.NetMode.IsBridge():
 		entries = etchosts.GetNetworkHostEntries(c.state.NetworkStatus, names...)
+	case c.config.NetMode.IsPasta():
+		var ip string
+		err := ns.WithNetNSPath(c.state.NetNS, func(_ ns.NetNS) error {
```

Comment:
Ah that is annoying, it is not compiling on freebsd, given pasta is not used there that is not a problem in the logic but the compiler cannot know that.
You have to move this block into a separate function in networking_linux.go for example and then create another one in networking_freebsd.go which juts return an error saying it is unsupported. Then you can call this function here and it should compile fine.

you can cross compile check locally with `make bin/podman.cross.freebsd.amd64`

Reply:
Thanks for the pointer, I think I've fixed this now.